### PR TITLE
SAMM CLI: Make native binaries easier to download via documentation

### PR DESCRIPTION
**Pull request by Yauhenikapl**
Changes:
- Update documentation. Were added links to the native binaries for Linux and Windows.

Fixes: #494

### CODE REVIEW

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 1]
```diff
@@ -5,10 +5,16 @@
 
 The SAMM CLI is a command line tool for the validation of Aspect Models and the generation of artifacts, such as documentation or code, from Aspect Models.
 
-TIP: Download latest version: icon:download[] https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}.jar[samm-cli-{esmf-sdk-version}.jar]
+The SAMM CLI offers multiple deployment options (you can download it bellow):
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:8]: The SAMM CLI is available as a native executable for multiple platforms and as an an executable jar:

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 2]
```diff
@@ -5,10 +5,16 @@
...
+
+TIP: Download latest version executable jar: icon:download[] https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-x86_64.jar[samm-cli-{esmf-sdk-version}-x86_64.jar]
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:10]: Download executable jar (latest):

- @atextor , @Yauhenikapl , I propose to use "native executable" instead of "native binary". Graal themselves use the wording (https://www.graalvm.org/latest/reference-manual/native-image/) and for "the basic windows user", "executable" carries "I can execute it" while "binary" can lead to ...oO(???)

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 3]
```diff
@@ -5,10 +5,16 @@
...
 
 NOTE: For technical reasons, Java 17 or newer is required to successfully run the tool.
 
+WINDOWS: Download latest version of native binaries for Windows: https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-windows-x86_64.zip[samm-cli-{esmf-sdk-version}-windows-x86_64.zip]
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:14]: "Download native executable for Windows (latest): "

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 4]
```diff
@@ -5,10 +5,16 @@
...
+
+LINUX: Download latest version of native binaries for Linux: https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz[samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz]
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:16]: "Download native executable for Linux (latest): "

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 5]
```diff
@@ -26,6 +32,18 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
 ----
 ====
 
+*For the Windows native binary:*
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:35]: native executable

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 6]
```diff
@@ -26,6 +32,18 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
...
+. *Download*: download above or visit our github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:36]: *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download the executable file for Windows.

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 7]
```diff
@@ -26,6 +32,18 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
...
+. *Extract*: extract it to a location of your choice.
+. *Open Command Prompt*: Launch the Command Prompt and navigate to the directory (samm-cli-windows) containing our application.
+. *Run*: In the command prompt, execute the command ``./samm.exe` to start the application.
+
+*For the Linux native binary:*
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:41]: native exectuable

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 8]
```diff
@@ -26,6 +32,18 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
...
+. *Download*: download above or visit our github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Linux.
+. *Extract*: Unpack the archive to your preferred directory.
+. *Open Terminal*: Open the Terminal and navigate to the directory with the application.
+. *Execute*: Run the application using the command `./samm`.
+
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:46]: . *Input*: Make sure to read the below documentation and provide model files in the correct directory structure

((perhaps link to https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html#models-directory-structure from the same page. This is a bit of duplication, but I fear people will at that point say "got it, don't need the documentation" and then get invalid models, particularly if it's only one and not in the right folder structure. This comment should be relaxed once the SAMM CLI has been made more convenient, see https://github.com/eclipse-esmf/esmf-sdk/issues/495 ))

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 9]
```diff
@@ -26,6 +32,18 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
...
 Successful execution of a command is signaled by returning 0. In case of logical or some other internal problem error code 1 is returned.
```
Comment (chris-volk) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:47]: In case of a logical or other internal error the error code 1 is being returned.

**Review by chris-volk — CHANGES_REQUESTED** [review]

File: .github/workflows/release-workflow.yml [hunk 10]
```diff
@@ -131,7 +133,7 @@ jobs:
           name: linux-artifacts
           path: |
             nexus-staging/
-            samm-cli-linux.tar.gz
+            samm-cli-${{ github.event.inputs.release_version }}-linux-x86_64.tar.gz
```
Comment (atextor) [inline .github/workflows/release-workflow.yml:136]: Can we use `$release_version` here too?

File: .github/workflows/release-workflow.yml [hunk 11]
```diff
@@ -131,7 +133,7 @@ jobs:
...
             samm-cli-*.jar
```
Comment (atextor) [inline .github/workflows/release-workflow.yml:137]: Please replace this line with the specific file names, so that we do not include the "samm-cli-${release_version}-for-native-image.jar" any more:

```
samm-cli-${release_version}-javadoc.jar
samm-cli-${release_version}-sources.jar
samm-cli-${release_version}-tests.jar
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 12]
```diff
@@ -5,10 +5,16 @@
 
 The SAMM CLI is a command line tool for the validation of Aspect Models and the generation of artifacts, such as documentation or code, from Aspect Models.
 
-TIP: Download latest version: icon:download[] https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}.jar[samm-cli-{esmf-sdk-version}.jar]
+The SAMM CLI is available as a native executable for multiple platforms and as an an executable jar:
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:8]:
```suggestion
The SAMM CLI is available as a native executable for multiple platforms and as an executable jar:
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 13]
```diff
@@ -5,10 +5,16 @@
...
+
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:9]: The `TIP:` and `NOTE:` are Asciidoc syntax to create these styled blocks with the icons:
image

Using `WINDOWS:` and `LINUX:` like this will look strange. I propose we put all download links as a list in the TIP section, i.e.:

```asciidoc
[TIP] 
.Downloading samm-cli
====
* Download executable for Windows (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-windows-x86_64.zip[samm-cli-{esmf-sdk-version}-windows-x86_64.zip]
* Download exectuable for Linux (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz[samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz]
* Download executable jar for other platforms and architectures (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}.jar[samm-cli-{esmf-sdk-version}.jar]. Running this version requires Java 17.
====
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 14]
```diff
@@ -26,7 +32,21 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
 ----
 ====
 
-Successful execution of a command is signaled by returning 0. In case of logical or some other internal problem error code 1 is returned.
+*For the Windows native executable:*
+. *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:36]:
```suggestion
. *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at the Github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 15]
```diff
@@ -26,7 +32,21 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
...
+. *Extract*: extract it to a location of your choice.
+. *Open Command Prompt*: Launch the Command Prompt and navigate to the directory (samm-cli-windows) containing our application.
+. *Input*: Make sure to read the below documentation and provide model files in the correct models directory structure[directory structure].
+. *Run*: In the command prompt, execute the command ``./samm.exe` to start the application.
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:40]:
```suggestion
. *Run*: In the command prompt, execute the command `samm` to start the application. See the below table for details.
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 16]
```diff
@@ -26,7 +32,21 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
...
+
+*For the Linux native executable:*
+. *Download*: download above or visit our github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Linux.
+. *Extract*: Unpack the archive to your preferred directory.
+. *Open Terminal*: Open the Terminal and navigate to the directory with the application.
+. *Input*: Make sure to read the below documentation and provide model files in the correct models directory structure[directory structure].
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:46]:
```suggestion
. *Input*: Make sure to read the below documentation and provide model files in the correct xref:models-directory-structure[directory structure].
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 17]
```diff
@@ -26,7 +32,21 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
...
-Successful execution of a command is signaled by returning 0. In case of logical or some other internal problem error code 1 is returned.
+*For the Windows native executable:*
+. *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.
+. *Extract*: extract it to a location of your choice.
+. *Open Command Prompt*: Launch the Command Prompt and navigate to the directory (samm-cli-windows) containing our application.
+. *Input*: Make sure to read the below documentation and provide model files in the correct models directory structure[directory structure].
+. *Run*: In the command prompt, execute the command ``./samm.exe` to start the application.
+
+*For the Linux native executable:*
+. *Download*: download above or visit our github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Linux.
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:43]:
```suggestion
. *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at the Github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Linux.
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 18]
```diff
@@ -26,7 +32,21 @@ alias samm='java -jar /location/to/samm-cli-{esmf-sdk-version}.jar'
 ----
 ====
 
-Successful execution of a command is signaled by returning 0. In case of logical or some other internal problem error code 1 is returned.
+*For the Windows native executable:*
+. *Download*: Download via above link or visit the repository of the Java SDK which contains the SAMM CLI at github https://github.com/eclipse-esmf/esmf-sdk/releases[releases page] and download executable file for Windows.
+. *Extract*: extract it to a location of your choice.
+. *Open Command Prompt*: Launch the Command Prompt and navigate to the directory (samm-cli-windows) containing our application.
+. *Input*: Make sure to read the below documentation and provide model files in the correct models directory structure[directory structure].
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:39]:
```suggestion
. *Input*: Make sure to read the below documentation and provide model files in the correct xref:models-directory-structure[directory structure].
```

File: tools/samm-cli/pom.xml
Comment (atextor) [inline tools/samm-cli/pom.xml:1]: Revert the changes in this file. The .jar is not architecture dependent and we should continue to call it "samm-cli-{version}.jar"

**Review by atextor — CHANGES_REQUESTED** [review]

File: .github/workflows/release-workflow.yml (hunk 10 above)
Comment (Yauhenikapl) [reply .github/workflows/release-workflow.yml:136]: Yes, sure. But here we have to make separate variable, I decided to don't this. I will add this.

Thanks!

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc (hunk 12 above)
Comment (Yauhenikapl) [reply documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:8]: Yes...
Thanks!

File: .github/workflows/release-workflow.yml [hunk 19]
```diff
@@ -131,8 +133,12 @@ jobs:
           name: linux-artifacts
           path: |
             nexus-staging/
-            samm-cli-linux.tar.gz
-            samm-cli-*.jar
+            samm-cli-${{ env.release_version }}-linux-x86_64.tar.gz
```
Comment (atextor) [inline .github/workflows/release-workflow.yml:136]: Since variable interpolation is error-prone, I would prefer to use `${{ github.event.inputs.release_version }}` everywhere. I.e. replace `${{ env.release_version }}` and `$release_version` by it. It's longer, but it will consistently work.

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 20]
```diff
@@ -5,9 +5,15 @@
...
-TIP: Download latest version: icon:download[] https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}.jar[samm-cli-{esmf-sdk-version}.jar]
+The SAMM CLI is available as a native executable for multiple platforms and as an executable jar:
 
-NOTE: For technical reasons, Java 17 or newer is required to successfully run the tool.
+[TIP]
+.Downloading samm-cli
+====
+* Download executable for Windows (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-windows-x86_64.zip[samm-cli-{esmf-sdk-version}-windows-x86_64.zip]
+* Download exectuable for Linux (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz[samm-cli-{esmf-sdk-version}-linux-x86_64.tar.gz]
+* Download executable jar for other platforms and architectures (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}-x86_64.jar[samm-cli-{esmf-sdk-version}-x86_64.jar]. Running this version requires Java 17.
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:15]:
```suggestion
* Download executable jar for other platforms and architectures (latest): https://github.com/eclipse-esmf/esmf-sdk/releases/download/v{esmf-sdk-version}/samm-cli-{esmf-sdk-version}.jar[samm-cli-{esmf-sdk-version}.jar]. Running this version requires Java 17.
```

**Review by atextor — CHANGES_REQUESTED** [review]

**Review by chris-volk — APPROVED** [review]

**Review by atextor — APPROVED** [review]
